src/CarbonAware.WebApi/src/Program.cs

```diff
@@ -32,6 +31,19 @@
 
 builder.Services.AddHealthChecks();
 
+var telemetryProvider = builder.Configuration["TELEMETRY_PROVIDER"];
```

Comment:
I wonder if it would be worth using a similar pattern to what we do for [DataSource Registration]( https://github.com/microsoft/carbon-aware-sdk/blob/dev/src/CarbonAware.DataSources/CarbonAware.DataSources.Registration/Configuration/ServiceCollectionExtensions.cs) to contain this logic.